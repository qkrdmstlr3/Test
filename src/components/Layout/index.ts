import {
  ShellHTML,
  createComponent,
  useGlobalState,
  RenderType,
} from '@Lib/shell-html';
import styleSheet from './style.scss';

class Layout extends ShellHTML {
  connectedCallback(): void {
    this.enrollObserving('page');
  }

  disconnectedCallback(): void {
    this.releaseObserving('page');
  }

  render(): RenderType {
    const pageName = useGlobalState('page');

    return {
      html: `
      <div class="layout">
        <header class="header">
          <span>slly</span>
        </header>
        <main class="main">
          <layout-nav></layout-nav>
          <page-${pageName}></page-${pageName}>
        </main>
      </div>
      `,
      css: styleSheet,
    };
  }
}

createComponent('layout-main', Layout);

export default Layout;
