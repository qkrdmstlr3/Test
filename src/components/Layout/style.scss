@import '@Style/variables.scss';

.main {
  height: 100%;
  background-color: $pink-color;
}

.header {
  width: 100%;
  height: 1.8rem;
  display: flex;
  justify-content: center;
  align-items: center;

  color: $deep-pink-color;
  background-color: $pink-color;
  font-weight: bold;
  font-size: 1.1rem;
  -webkit-app-region: drag;
}

.content {
  width: calc(100% - 5rem);
  border-top-left-radius: 10px;
  background-color: $light-pink-color;
}

.body {
  width: 100%;
  height: 100%;
  display: flex;
}
