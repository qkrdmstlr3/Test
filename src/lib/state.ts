import { state } from '@Lib/shell-html';

state({
  key: 'page',
  initial: 'home',
});

state({
  key: 'checklist',
  initial: [],
});

state({
  key: 'checkpostControl',
  initial: {
    currentCheckListId: undefined,
    currentCheckPostId: undefined,
  },
});

state({
  key: 'checkposts',
  initial: [],
});
