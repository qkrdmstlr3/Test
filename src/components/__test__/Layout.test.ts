import { getByText } from '@testing-library/dom';
import '@testing-library/jest-dom/extend-expect';
import Layout from '../Layout';
import '@Lib/state';

let layout: Layout;
let layoutComponent: HTMLElement;

describe('Layout Component test', () => {
  beforeEach(() => {
    layout = new Layout();
    if (layout.shadowRoot?.childNodes[0]) {
      layoutComponent = (layout.shadowRoot as unknown) as HTMLElement;
    }
  });

  it('title 확인', () => {
    if (!layoutComponent) return;

    const slly = getByText(layoutComponent, 'slly');
    expect(slly.innerHTML).toBe('slly');
  });

  it('초기 페이지가 home 페이지인지 확인', () => {
    if (!layoutComponent) return;

    expect(layoutComponent.innerHTML).toContain('<page-home></page-home>');
  });

  it('snapshot', () => {
    expect(layoutComponent).toMatchSnapshot();
  });
});
