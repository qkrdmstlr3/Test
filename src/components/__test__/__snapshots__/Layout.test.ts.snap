// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component test snapshot 1`] = `
ShadowRoot {
  Symbol(SameObject caches): Object {
    "childNodes": NodeList [
      <div
        class="layout"
      >
        <header
          class="header"
        >
          <span>
            slly
          </span>
        </header>
        <main
          class="main"
        >
          <layout-nav />
          <page-home />
        </main>
      </div>,
      <style>
        [object Object]
      </style>,
    ],
  },
}
`;
